ProcessExporter.py

```diff
@@ -356,63 +354,63 @@ def get_constructor_lines(self, matrix_element, color_amplitudes):
 
     def get_finalstates_map(self):
         """Build map of final states (instanciates SubProcess class)  """
-        
+
         final_states = {}
-       
+
         # First retrieve all the final state's SubProcess definitions
 
         for me in self.matrix_elements:
-            
+
             proc = me.get('processes')[0]
-            
+
             final_ids = "{" + ",".join( [ str(i) for i in proc.get_final_ids_after_decay() ] ) + "}"
-            
+
             iproc = {}
-            
-            iproc["function"] = "&%s::matrix_%s" % (self.process_class, proc.shell_string().replace("0_", ""))
+
+            iproc["function"] = "&{}::matrix_{}".format(self.process_class, proc.shell_string().replace("0_", ""))
             iproc["mirror"] = ( me.get('has_mirror_process') and "true" ) or "false"
             iproc["istates"] = "{" + ",".join( [ "std::make_pair(%i, %i)" % (proc.get('legs')[0].get('id'), proc.get('legs')[1].get('id')) for proc in me.get('processes') ] ) + "}"
-            iproc["ncomb"] = me.get_helicity_combinations() 
-            iproc["denom"] = me.get_denominator_factor() 
-            
+            iproc["ncomb"] = me.get_helicity_combinations()
+            iproc["denom"] = me.get_denominator_factor()
+
             final_states[final_ids] = final_states.get(final_ids, []) + [iproc]
 
         # Then define the actual final states map using these
 
         out  = ""
-        
-        for final, data in final_states.items():
-            
+
+        for final, data in list(final_states.items()):
```

Comment:
Is that conversion to list really needed?

Reply:
Turns out no: https://stackoverflow.com/questions/17695456/why-does-python-3-need-dict-items-to-be-wrapped-with-list

`2to3` is just overly cautious here, so I've manually changed it after `2to3` runs in the rebase.